### ISSUE TITLE
Dependencies added to indirect rather than direct when package is installed

### ISSUE THREAD

**Issue by jamesrweb** [wolfadex/elm-open-api#7]
When installing this package it puts `elm-community/json-extra`, `elm/http` and `elm/json` as indirect dependencies still, even though in the generated output these are required and should be "direct" instead.

**Comment by wolfadex** [comment]
Those are indirect dependencies of this package, which is accurate. They are however needed if you're using this in conjunction with the CLI codegen tool. If you were using this for say generating documentation then you wouldn't want them as direct dependencies.

Related though, the first of them is an issue here https://github.com/wolfadex/elm-open-api-cli/issues/63. The other 2 I'm not sure how I'd fix that for someone using the CLI codegen tool

**Comment by jamesrweb** [comment]
Perhaps just documenting it in the READMEs and in the cli tool output to say "if I did X, the consumer will need Y and so I will inform them at the 'successful' output step." - kind of like when you create a new project with `vite` and it then says that they recommend to `cd` and `install` the dependencies into the directory and so on.

Even better would be if the generator checked "does the user have these dependencies installed" and if one or more are not, inform the user about that and then install the dependencies for them if they opt-in, otherwise inform them what to run to install them if they opt-out.

Do these kind of things make sense? Just spitballing some ideas. Let me know what you think and we can iterate together.

**Comment by wolfadex** [comment]
Can definitely add a note in the docs here and in the CLI.

Something like this (The `SDK generated at ...` part is the current message when the SDK successfully builds.) if those deps aren't detected? That should be possible I think. I know there's a package for parsing `elm.json`, but it might be faster to do a simple decoder and search for those 2 deps.
```sh
SDK generated at some/path.elm

You'll need elm/http and elm/json installed. Try running:

    elm install elm/http
    elm install elm/json
```

That should be fairly easy to [add here](https://github.com/wolfadex/elm-open-api-cli/blob/67063684706a15ded606b3e53baa1705acabf12d/src/Cli.elm#L194) in the CLI code. Though I'd want to make sure that it fails gracefully since the generator has succeeded at this point.

**Comment by jamesrweb** [comment]
wolfadex/elm-open-api-cli#66 resolves the extra install instruction steps required for successful outputs. It isn't ideal since it will need maintained over time if more requirements become necessary since this cannot be dynamic AFAIK, unless we track explicitly what is used during generation and pass it as another parameter to the lambda for dynamic string generation instead, which would be arguably better in the long term 🤷🏻.

That just leaves the issue with incorrect outputs from "successful" generations but that one I shall leave to yourself and @miniBill since I amn't sure how, where or what to begin on that one. If I can support though, do let me know and I will see what I can do.

**Comment by jamesrweb** [comment]
@wolfadex why was this closed when the second point is still outstanding?

**Comment by wolfadex** [comment]
@jamesrweb which second point? This issue was for the dependencies correct? That is resolved in [v0.2.2](https://www.npmjs.com/package/elm-open-api/v/0.2.2).

**Comment by jamesrweb** [comment]
> @jamesrweb which second point? This issue was for the dependencies correct? That is resolved in [v0.2.2](https://www.npmjs.com/package/elm-open-api/v/0.2.2).
> 
> 

I meant the one in wolfadex/elm-open-api-cli#65 but mistakenly thought it was part of this issue, my bad! Sorry!
